### PR TITLE
Add error code for ScanOperationStates table

### PR DESCRIPTION
**Pull request by erdembayar**
Addresses: https://github.com/NuGet/Engineering/issues/5250
Adding 2 new columns to persist extra details about scan operation

### CODE REVIEW

**Review by drewgillies — DISMISSED** [review]

File: src/NuGet.Services.Validation/Entities/ScanOperationState.cs [hunk 1]
```diff
@@ -66,6 +66,16 @@ public class ScanOperationState
         /// </summary>
         public string OperationId { get; set; }
 
+        /// <summary>
+        /// Operation errors code.
+        /// </summary>
+        public int ErrorCode { get; set; }
```
Comment (joelverhagen) [inline src/NuGet.Services.Validation/Entities/ScanOperationState.cs:72]: does this actually need to be an int? Would a string be more flexible and get the job done?

Comment (erdembayar) [reply src/NuGet.Services.Validation/Entities/ScanOperationState.cs]: Sure, addressed it.

File: src/NuGet.Services.Validation/Entities/ScanOperationState.cs [hunk 2]
```diff
@@ -66,6 +66,16 @@ public class ScanOperationState
         /// </summary>
         public string OperationId { get; set; }
 
+        /// <summary>
+        /// Operation errors code.
```
Comment (agr) [inline src/NuGet.Services.Validation/Entities/ScanOperationState.cs:70]:
```suggestion
        /// Operation error code.
```

File: src/NuGet.Services.Validation/Entities/ScanOperationState.cs [hunk 3]
```diff
@@ -66,6 +66,16 @@ public class ScanOperationState
...
+        /// </summary>
+        public string ErrorCode { get; set; }
+
+        /// <summary>
+        /// Any applicable operation details in text/json format.
```
Comment (agr) [inline src/NuGet.Services.Validation/Entities/ScanOperationState.cs:75]: "text/json" here means "plain text or json" as far as I understand, but can be interpreted as MIME type string. I'd spell it out more explicitly to not create any confusion about potential content.

**Review by agr — APPROVED** [review]
Please, fix comments. Otherwise, looks good.

**Review by agr — APPROVED** [review]

**Pull request merged**